src/main/java/io/stargate/sgv2/jsonapi/service/processor/CommandProcessor.java

```diff
@@ -46,36 +60,92 @@ public CommandProcessor(
-    return commandResolverService
-        .resolverForCommand(command)
-
-        // resolver can be null, not handled in CommandResolverService for now
-        .flatMap(
-            resolver -> {
-              // if we have resolver, resolve operation and execute
-              Operation operation = resolver.resolveCommand(commandContext, command);
-              return operation.execute(queryExecutor);
-            })
+    final Tag commandTag = Tag.of(MetricsConstants.COMMAND, command.getClass().getSimpleName());
+    final String tenant =
+        stargateRequestInfo != null && stargateRequestInfo.getTenantId().isPresent()
+            ? stargateRequestInfo.getTenantId().get()
+            : MetricsConstants.UNKNOWN_VALUE;
+    final Tag tenantTag = Tag.of(MetricsConstants.TENANT, tenant);
+    final long start = System.currentTimeMillis();
+    return Uni.createFrom()
+        .item(start)
+        .onItem()
+        .transformToUni(
+            startTime -> {
+              // start by resolving the command, get resolver
+              return commandResolverService
+                  .resolverForCommand(command)
 
-        // handler failures here
-        .onFailure()
-        .recoverWithItem(
-            t -> {
-              // DocsException is supplier of the CommandResult
-              // so simply return
-              if (t instanceof JsonApiException jsonApiException) {
-                return jsonApiException;
-              }
+                  // resolver can be null, not handled in CommandResolverService for now
+                  .flatMap(
+                      resolver -> {
+                        // if we have resolver, resolve operation and execute
+                        Operation operation = resolver.resolveCommand(commandContext, command);
+                        return operation.execute(queryExecutor);
+                      })
 
-              // otherwise use generic for now
-              return new ThrowableCommandResultSupplier(t);
-            })
+                  // handler failures here
+                  .onItemOrFailure()
+                  .transform(
+                      (item, failure) -> {
+                        if (failure != null) {
+                          if (failure instanceof JsonApiException jsonApiException) {
+                            return jsonApiException;
+                          }
+                          return new ThrowableCommandResultSupplier(failure);
 
-        // if we have a non-null item
-        // call supplier get to map to the command result
-        .onItem()
-        .ifNotNull()
-        .transform(Supplier::get);
+                        } else {
+                          // return the tags and command result
+                          return item;
+                        }
+                      })
+                  .onItem()
+                  .ifNotNull()
+                  .transform(Supplier::get)
```

Comment:
I am not sure why you changed all the code above, when you anyway inspect the results from the final result?